src/go/k8s/controllers/redpanda/cluster_controller.go

```diff
@@ -256,17 +258,9 @@ func (r *ClusterReconciler) Reconcile(
 		secrets = append(secrets, schemaRegistrySu.Key())
 	}
 
-	err = r.setInitialSuperUserPassword(ctx, adminAPI, secrets)
-
-	var e *resources.RequeueAfterError
-	if errors.As(err, &e) {
-		log.Info(e.Error())
-		return ctrl.Result{RequeueAfter: e.RequeueAfter}, nil
-	}
-
-	if err != nil {
-		log.Error(err, "Failed to set up initial super user password")
-		return ctrl.Result{}, err
+	if errSetInit := r.setInitialSuperUserPassword(ctx, adminAPI, secrets); errSetInit != nil {
+		log.Info(errSetInit.Error())
+		return ctrl.Result{RequeueAfter: resources.RequeueDuration}, errSetInit
```

Comment:
if you are returning an error the RequeueAfter is not taken into consideration by controller runtime 

```go
	// RunInformersAndControllers the syncHandler, passing it the Namespace/Name string of the
	// resource to be synced.
	result, err := c.Reconcile(ctx, req)
	switch {
	case err != nil:
		c.Queue.AddRateLimited(req)
		ctrlmetrics.ReconcileErrors.WithLabelValues(c.Name).Inc()
		ctrlmetrics.ReconcileTotal.WithLabelValues(c.Name, labelError).Inc()
		log.Error(err, "Reconciler error")
	case result.RequeueAfter > 0:
		// The result.RequeueAfter request will be lost, if it is returned
		// along with a non-nil error. But this is intended as
		// We need to drive to stable reconcile loops before queuing due
		// to result.RequestAfter
		c.Queue.Forget(obj)
		c.Queue.AddAfter(req, result.RequeueAfter)
		ctrlmetrics.ReconcileTotal.WithLabelValues(c.Name, labelRequeueAfter).Inc()
	case result.Requeue:
		c.Queue.AddRateLimited(req)
		ctrlmetrics.ReconcileTotal.WithLabelValues(c.Name, labelRequeue).Inc()
	default:
		// Finally, if no error occurs we Forget this item so it does not
		// get queued again until another change happens.
		c.Queue.Forget(obj)
		ctrlmetrics.ReconcileTotal.WithLabelValues(c.Name, labelSuccess).Inc()
	}
```
REF:
https://github.com/kubernetes-sigs/controller-runtime/blob/53057ba616d1ba05ea6e295c1da870a555d9988d/pkg/internal/controller/controller.go#L312-L337

Reply:
Yes, realized this error recently.